AlfrescoKit/AlfrescoKit/Picking View Controller/AKAlfrescoNodePickingListViewController.m

```diff
@@ -0,0 +1,258 @@
+                                               session:(id<AlfrescoSession>)session
+{
+    self = [self initWithAlfrescoFolder:folder listingContext:listingContext session:session];
+    if (self)
+    {
+        self.nodePickerOptions = AKAlfrescoNodePickerTypeFolderPicker;
+        self.delegate = delegate;
+        self.selectedNodes = (selectedNodes) ? selectedNodes : [NSMutableArray array];
+    }
+    return self;
+}
+
+- (instancetype)initAlfrescoDocumentPickerWithRootFolder:(AlfrescoFolder *)folder
+                                       multipleSelection:(BOOL)allowMultiple
+                                           selectedNodes:(NSMutableArray *)selectedNodes
+                                                delegate:(id<AKAlfrescoNodePickingListViewControllerDelegate>)delegate
+                                                 session:(id<AlfrescoSession>)session
+{
+    return [self initAlfrescoDocumentPickerWithRootFolder:folder multipleSelection:allowMultiple selectedNodes:selectedNodes delegate:delegate listingContext:nil session:session];
+}
+
+- (instancetype)initAlfrescoDocumentPickerWithRootFolder:(AlfrescoFolder *)folder
+                                       multipleSelection:(BOOL)allowMultiple
+                                           selectedNodes:(NSMutableArray *)selectedNodes
+                                                delegate:(id<AKAlfrescoNodePickingListViewControllerDelegate>)delegate
+                                          listingContext:(AlfrescoListingContext *)listingContext
+                                                 session:(id<AlfrescoSession>)session
+{
+    self = [self initWithAlfrescoFolder:folder listingContext:listingContext session:session];
+    if (self)
+    {
+        self.nodePickerOptions = AKAlfrescoNodePickerTypeDocumentPicker;
+        if (allowMultiple)
+        {
+            self.nodePickerOptions |= AKAlfrescoNodePickerTypeMultiplePicker;
+        }
+        self.delegate = delegate;
+        self.selectedNodes = (selectedNodes) ? selectedNodes : [NSMutableArray array];
+        
+    }
+    return self;
+}
+
+- (NSString *)nibName
+{
+    return NSStringFromClass([self superclass]);
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    // Setup the toolbar
+    [self setupToolbar];
+    // Display the toolbar if required
+    BOOL isFolderPicker = self.nodePickerOptions & AKAlfrescoNodePickerTypeFolderPicker;
+    BOOL isMultipleDocumentPicker = self.nodePickerOptions & AKAlfrescoNodePickerTypeDocumentPicker && self.nodePickerOptions & AKAlfrescoNodePickerTypeMultiplePicker;
+    
+    if (isFolderPicker || isMultipleDocumentPicker)
+    {
+        self.navigationController.toolbarHidden = NO;
+    }
+}
+
+#pragma mark - Private Methods
+
+- (void)chooseButtonSelected:(id)sender
+{
+    if (self.nodePickerOptions & AKAlfrescoNodePickerTypeFolderPicker)
+    {
+        [self.delegate nodePickingListViewController:self didSelectNodes:@[_folder]];
```

Comment:
Should probably try to avoid `_folder` direct access unless it's for a good reason (which should be commented)